### PR TITLE
Adds SPIFFS to the larger nucleo boards

### PR DESCRIPTION
**Pull request by balazsracz**
Refactors SPIFFS so that we don't need to include spffs.h in SPIFFS.hxx. This allows compiling the target HwInit.cxx without needing to worry about -I arguments to find spiffs source.
Some other minor fixes to drivers.

### CODE REVIEW

File: src/freertos_drivers/sources [hunk 1]
```diff
@@ -51,6 +52,7 @@ endif
 ifeq ($(TARGET),freertos.armv7m.exc)
 
 SUBDIRS += stm32cubef303x_28x_58x_98x stm32cubef303xe stm32cubef767xx \
+         spiffs_stm32f303xe spiffs_stm32f767xx \
```
Comment (bakerstu) [inline src/freertos_drivers/sources]: Alignment with line above, add two spaces to the indent.

File: src/freertos_drivers/spiffs/SPIFFS.cxx [hunk 2]
```diff
@@ -93,7 +131,66 @@ SPIFFS::SPIFFS(size_t physical_address, size_t size_on_disk,
     , formatted_(false)
     , anyDirty_(false)
 {
-    fs_.user_data = this;
+    fs_ = new spiffs;
+    fs_->user_data = this;
+    spiffs_config tmp{.hal_read_f       = flash_read,
+               .hal_write_f      = flash_write,
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/SPIFFS.cxx]: alignment with line above.

File: src/freertos_drivers/spiffs/SPIFFS.cxx [hunk 3]
```diff
@@ -45,6 +46,20 @@
 #define O_DIRECT _FDIRECT
 #endif
 
+/// Provide mutex lock.
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/SPIFFS.cxx]: All of the Doxygen comment header blocks for function headers should go in their respective header files, if applicable, and not in the the *.cxx file.  Keep them all in one place.

File: src/freertos_drivers/spiffs/SPIFFS.cxx [hunk 4]
```diff
@@ -45,6 +46,20 @@
...
+/// @param fs reference to the file system instance
+void SPIFFS::extern_lock(struct spiffs_t *fs)
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/SPIFFS.cxx]: These "extern_lock" functions seem to be declared "inline" in the header file.  Don't these have to be implemented as inline in the header file as well?

File: src/freertos_drivers/spiffs/SPIFFS.hxx [hunk 5]
```diff
@@ -204,16 +174,12 @@ protected:
     /// @param size size of erase region in bytes
     virtual int32_t flash_erase(uint32_t addr, uint32_t size) = 0;
 
-    /// configuration parameters for SPIFFS
-    spiffs_config config_;
-
+    /// file system instance metadata
+    spiffs* fs_;
+
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/SPIFFS.hxx]: extra whitespace.

File: src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file Stm32SPIFFS.cxx
+ * This file implements a SPIFFS FLASH driver specific to CC32xx.
+ *
+ * @author Stuart W. Baker
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx]: This and other new files should be attributed to Balazs as the author and copyright holder, also update the dates.

File: src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx [hunk 7]
```diff
@@ -0,0 +1,143 @@
+/** @copyright
+ * Copyright (c) 2018, Stuart W Baker
```
Comment (bakerstu) [inline src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx]: This and other new files should be attributed to Balazs as the author and copyright holder, also update the dates.

File: src/freertos_drivers/sources (hunk 1 above)
Comment (balazsracz) [reply src/freertos_drivers/sources]: done

File: src/freertos_drivers/spiffs/SPIFFS.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/SPIFFS.cxx]: done

File: src/freertos_drivers/spiffs/SPIFFS.cxx (hunk 3 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/SPIFFS.cxx]: all cleaned up in this file.

File: src/freertos_drivers/spiffs/SPIFFS.cxx (hunk 4 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/SPIFFS.cxx]: Technically their definition must be available in the compilation unit where they are used. That's only in SPIFFS.cxx for this case.
The assembly code generated is as follows:
```
0803f5e0 <extern_spiffs_lock>:
 803f5e0:	b508      	push	{r3, lr}
 803f5e2:	6f80      	ldr	r0, [r0, #120]	; 0x78
 803f5e4:	3024      	adds	r0, #36	; 0x24
 803f5e6:	f7d2 facf 	bl	8011b88 <OSMutex::lock()>
 803f5ea:	bd08      	pop	{r3, pc}
```
which shows that inlining happened correctly.

File: src/freertos_drivers/spiffs/SPIFFS.hxx (hunk 5 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/SPIFFS.hxx]: done

File: src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx (hunk 6 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx]: done

File: src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx (hunk 7 above)
Comment (balazsracz) [reply src/freertos_drivers/spiffs/stm32f0_f3/Stm32SPIFFS.cxx]: done

**Pull request merged**